src/java/org/apache/cassandra/io/sstable/format/big/BigTableScanner.java

```diff
@@ -345,11 +339,15 @@ protected UnfilteredRowIterator computeNext()
 
                 /*
                  * For a given partition key, we want to avoid hitting the data
-                 * file unless we're explicitely asked to. This is important
+                 * file unless we're explicitly asked to. This is important
                  * for PartitionRangeReadCommand#checkCacheFilter.
                  */
                 return new LazilyInitializedUnfilteredRowIterator(currentKey)
                 {
+                    // Store currentEntry reference during object instantiation as later (during initialize) the
+                    // reference may point to a different entry.
+                    private final RowIndexEntry rowIndexEntry = currentEntry;
```

Comment:
Was there are test failing somewhere around this, or is it just something we noticed offhand?

Reply:
Test for the associated failure plus proper fix added.